types/vscode-webview/index.d.ts

```diff
@@ -0,0 +1,50 @@
+// Type definitions for non-npm package vscode-webview 1.57
+// Project: https://code.visualstudio.com/api/extension-guides/webview
+// Definitions by: Matt Bierner <https://github.com/mjbvz>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.0
+
+/**
+ * API exposed to webviews.
+ *
+ * @template StateType Type of the persisted state stored for the webview.
+ */
+export interface WebviewApi<StateType> {
+    /**
+     * Post a message to the owner of the webview.
+     *
+     * @param message Data to post. Must be JSON serializable.
+     */
+    postMessage(message: any): void;
+
+    /**
+     * Get the persistent state stored for this webview.
+     *
+     * @return The current state or `undefined` if no state has been set.
+     */
+    getState(): StateType | undefined;
+
+    /**
+     * Set the persistent state stored for this webview.
+     *
+     * @param newState New persisted state. This must be a JSON serializable object. Can be retrieved
+     * using {@link getState}.
+     *
+     * @return The new state.
+     */
+    setState(newState: StateType): StateType;
+    setState(newState: undefined): undefined;
```

Comment:
```suggestion
    setState<T extends StateType>(newState: T): T;
```

is more accurate if you want to represent return-what-is-given